iceoryx2-ffi/ffi/src/node.rs

```diff
@@ -0,0 +1,348 @@
+    INTERNAL_ERROR,
+}
+
+impl IntoCInt for NodeCreationFailure {
+    fn into_c_int(self) -> c_int {
+        (match self {
+            NodeCreationFailure::InsufficientPermissions => {
+                iox2_node_creation_failure_e::INSUFFICIENT_PERMISSIONS
+            }
+            NodeCreationFailure::InternalError => iox2_node_creation_failure_e::INTERNAL_ERROR,
+        }) as c_int
+    }
+}
+
+#[repr(C)]
+#[repr(align(8))] // alignment of NodeBuilder
+pub struct iox2_node_builder_storage_internal_t {
+    internal: [u8; 18416], // magic number obtained with size_of::<NodeBuilder>()
+}
+
+#[repr(C)]
+pub struct iox2_node_builder_storage_t {
+    internal: iox2_node_builder_storage_internal_t,
+    deleter: fn(*mut iox2_node_builder_storage_t),
+}
+
+/// The handle to use for the `iox2_node_builder_*` functions
+pub type iox2_node_builder_h = *mut iox2_node_builder_storage_t;
+
+impl iox2_node_builder_storage_t {
+    fn node_builder_maybe_uninit(&mut self) -> &mut MaybeUninit<NodeBuilder> {
+        static_assert_gt_or_equal::<
+            { align_of::<iox2_node_builder_storage_internal_t>() },
+            { align_of::<NodeBuilder>() },
+        >();
+        static_assert_gt_or_equal::<
+            { size_of::<iox2_node_builder_storage_internal_t>() },
+            { size_of::<NodeBuilder>() },
+        >();
+
+        unsafe {
+            &mut *(&mut self.internal as *mut iox2_node_builder_storage_internal_t)
+                .cast::<MaybeUninit<NodeBuilder>>()
+        }
+    }
+    unsafe fn node_builder_assume_init(&mut self) -> &mut NodeBuilder {
+        self.node_builder_maybe_uninit().assume_init_mut()
+    }
+
+    fn alloc() -> *mut iox2_node_builder_storage_t {
+        unsafe {
+            alloc(Layout::new::<iox2_node_builder_storage_t>()) as *mut iox2_node_builder_storage_t
+        }
+    }
+    fn dealloc(storage: *mut iox2_node_builder_storage_t) {
+        unsafe {
+            dealloc(
+                storage as *mut _,
+                Layout::new::<iox2_node_builder_storage_t>(),
+            );
+        }
+    }
+}
+
+// END iox2_node_builder_* types
+
+// BEGIN iox2_builder_* types
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum iox2_node_type_e {
+    PROCESS_LOCAL,
+    ZERO_COPY,
+}
+
+#[repr(C)]
+#[repr(align(8))] // magic number; the larger one of align_of::<Node<zero_copy::Service>>() and align_of::<Node<process_local::Service>>()
+pub struct iox2_node_storage_internal_t {
+    internal: [u8; 8], // magic number; the larger one of size_of::<Node<zero_copy::Service>>() and size_of::<Node<process_local::Service>>()
+}
+
+#[repr(C)]
+pub struct iox2_node_storage_t {
+    node_type: iox2_node_type_e,
+    internal: iox2_node_storage_internal_t,
+    deleter: fn(*mut iox2_node_storage_t),
+}
+
+/// The handle to use for the `iox2_node_*` functions
+pub type iox2_node_h = *mut iox2_node_storage_t;
+
+impl iox2_node_storage_t {
+    const fn max(a: usize, b: usize) -> usize {
```

Comment:
Could you move this into `iceoryx2_bb_elementary/math.rs`. A `const` max is something that could be useful on other places as well.

And could you add a little test as well that test the case `a<b`, `a==b` and `a>b`